### ISSUE TITLE
CacheStampedeBlocker JSONObject exception

### ISSUE THREAD

**Issue by valentijnscholten** [DependencyTrack/dependency-track#2456]
### Current Behavior

Logs contain lots of errors trying to populate metadata cache:
```
2023-02-04 10:54:14,609 ERROR [CacheStampedeBlocker] An error occurred while populating cache repositoryMetaCache for key pkg:composer/magento/module-developer@100.4.5 : JSONObject["magento/module-developer"] not found.
```

### Steps to Reproduce

Upload some SBOMS with magento modules/components such as `magento/module-developer`.
Wait for Metadata update jobs to complete
Observe log

### Expected Behavior

It looks like its a common scenario to not being able to retrieve metadata for some components, so this should be handled more gracefully. Maybe lower to WARN or even INFO/DEBUG. Similar to what was done in #2134
Also currently the details of the exception are swallowed. 
I haven't deduced yet the exact spot where this JSONObject not found error occurs.
There are 2 places where this log line is generated:

https://github.com/DependencyTrack/dependency-track/blob/836da60abf648e3fc091d724aef3b3d0071ba7b4/src/main/java/org/dependencytrack/util/CacheStampedeBlocker.java#L193-L200

https://github.com/DependencyTrack/dependency-track/blob/836da60abf648e3fc091d724aef3b3d0071ba7b4/src/main/java/org/dependencytrack/util/CacheStampedeBlocker.java#L216-L219

### Dependency-Track Version

4.7.1

### Dependency-Track Distribution

Container Image

### Database Server

PostgreSQL

### Browser

Google Chrome

### Checklist

- [X] I have read and understand the [contributing guidelines](https://github.com/DependencyTrack/dependency-track/blob/master/CONTRIBUTING.md#filing-issues)
- [X] I have checked the [existing issues](https://github.com/DependencyTrack/dependency-track/issues) for whether this defect was already reported

**Comment by nscuro** [comment]
I remember we had a Slack discussion about properly handling 404s / "package not found" cases instead of letting them run into the default exception handling. We should definitely get this sorted. Thanks for reporting.

**Comment by valentijnscholten** [comment]
should maybe be en ehancement request instead of defect.

**Comment by valentijnscholten** [comment]
I think this will be resolved by #2327 which is not yet released. I was confused because A) I thought the fix was already released B) In 4.7.x the exception in a different class which didn't sound familiar.

**Comment by nscuro** [comment]
#2327 only addressed Composer, not any of the other repo analyzers. So still a valid issue.

**Comment by valentijnscholten** [comment]
The reason I became aware of this is because this exception caused excessive retries to be done by the `CacheStampedeBlocker` saturating the worker threads for a long time.

**Comment by nscuro** [comment]
Resolved in #2563
